ext/uri/php_lexbor.c

```diff
@@ -73,18 +73,20 @@ static void lexbor_cleanup_parser(void)
  * When errors is NULL, the caller is not interested in the additional error information,
  * so the function does nothing.
  */
-static void fill_errors(zval *errors)
+static zend_string *fill_errors(zval *errors)
 {
+	zend_string *result = NULL;
```

Comment:
What we try to do nowadays is to declare variables closer to their actual usage (at least I definitely did this in ext/uri). `result` could go around line 91, and these returns could be simply `NULL` so that one wouldn't need to check what the current value of `result` is.